leshan-integration-tests/src/test/java/org/eclipse/leshan/integration/tests/util/QueuedModeLeshanClient.java

```diff
@@ -0,0 +1,202 @@
+/*******************************************************************************
+ * Copyright (c) 2015 Bosch Software Innovations GmbH and others.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * and Eclipse Distribution License v1.0 which accompany this distribution.
+ *
+ * The Eclipse Public License is available at
+ *    http://www.eclipse.org/legal/epl-v10.html
+ * and the Eclipse Distribution License is available at
+ *    http://www.eclipse.org/org/documents/edl-v10.html.
+ *
+ * Contributors:
+ *     Alexander Ellwein (Bosch Software Innovations GmbH)
+ *                     - initial API and implementation
+ *******************************************************************************/
+package org.eclipse.leshan.integration.tests.util;
+
+import org.eclipse.californium.core.CoapServer;
+import org.eclipse.californium.core.network.CoapEndpoint;
+import org.eclipse.californium.core.network.config.NetworkConfig;
+import org.eclipse.californium.core.server.resources.CoapExchange;
+import org.eclipse.californium.core.server.resources.Resource;
+import org.eclipse.californium.scandium.DTLSConnector;
+import org.eclipse.californium.scandium.config.DtlsConnectorConfig;
+import org.eclipse.leshan.LwM2mId;
+import org.eclipse.leshan.client.LwM2mClient;
+import org.eclipse.leshan.client.californium.impl.*;
+import org.eclipse.leshan.client.resource.LwM2mObjectEnabler;
+import org.eclipse.leshan.client.servers.BootstrapHandler;
+import org.eclipse.leshan.client.servers.RegistrationEngine;
+import org.eclipse.leshan.util.Validate;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.net.InetSocketAddress;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+import java.util.concurrent.ConcurrentHashMap;
+
+/**
+ * This client has a special ability to have its GET handler instrumented by a test by which different client
+ * scenarios like sleep, wake up can be simulated.
+ */
+public class QueuedModeLeshanClient implements LwM2mClient {
```

Comment:
This is used only for testing purposes so that we can set the client to or not to respond to GET. It is not part of the leshan-server-core.